hikari/api/cache.py

```diff
@@ -78,6 +79,17 @@ class Cache(abc.ABC):
 
     __slots__: typing.Sequence[str] = ()
 
+    @property
+    @abc.abstractmethod
+    def settings(self) -> config.CacheSettings:
```

Comment:
split into base config with only the flags and then an extended one with client specific data for specific resources

Reply:
I'ma do this in another PR